tachyon/math/polynomial/domains/mixed_radix/mixed_radix_evaluation_domain.h

```diff
@@ -0,0 +1,356 @@
+// Copyright 2022 arkworks contributors
+// Use of this source code is governed by a MIT/Apache-2.0 style license that
+// can be found in the LICENSE-MIT.arkworks and the LICENCE-APACHE.arkworks
+// file.
+
+// This header contains a |MixedRadixEvaluationDomain| for performing various
+// kinds of polynomial arithmetic on top of fields that are FFT-friendly but do
+// not have high-enough two-adicity to perform the FFT efficiently, i.e. the
+// multiplicative subgroup G generated by |F::Config::kTwoAdicRootOfUnity| is
+// not large enough. |MixedRadixEvaluationDomain| resolves this issue by using a
+// larger subgroup obtained by combining G with another subgroup of size
+// |F::Config::kSmallSubgroupBase|^|F::Config::kSmallSubgroupAdicity|, to obtain
+// a subgroup generated by |F::Config::kLargeSubgroupRootOfUnity|.
+
+#ifndef TACHYON_MATH_POLYNOMIAL_DOMAINS_MIXED_RADIX_MIXED_RADIX_EVALUATION_DOMAIN_H_
+#define TACHYON_MATH_POLYNOMIAL_DOMAINS_MIXED_RADIX_MIXED_RADIX_EVALUATION_DOMAIN_H_
+
+#include <stddef.h>
+#include <stdint.h>
+
+#include "tachyon/base/logging.h"
+#include "tachyon/base/strings/string_number_conversions.h"
+#include "tachyon/math/base/gmp/gmp_util.h"
+#include "tachyon/math/finite_fields/prime_field_util.h"
+#include "tachyon/math/polynomial/domains/evaluation_domain.h"
+#include "tachyon/math/polynomial/domains/evaluation_domain_util.h"
+
+namespace tachyon::math {
+
+// Defines a domain over which finite field (I)FFTs can be performed. Works only
+// for fields that have a multiplicative subgroup of size that is a power-of-2
+// and another small subgroup over a different base defined.
+template <typename F>
+class MixedRadixEvaluationDomain
+    : public EvaluationDomain<MixedRadixEvaluationDomain<F>> {
+ public:
+  using Field = F;
+
+  constexpr MixedRadixEvaluationDomain() = default;
+  // Construct a domain that is large enough for evaluations of a polynomial
+  // having |num_coeffs| coefficients.
+  constexpr explicit MixedRadixEvaluationDomain(size_t num_coeffs) {
+    CHECK(F::Config::kHasLargeSubgroupRootOfUnity);
+
+    // Compute the best size of our evaluation domain.
+    size_ = BestMixedDomainSize(num_coeffs);
+
+    // Compute the size of our evaluation domain
+    uint64_t q = static_cast<uint64_t>(Field::Config::kSmallSubgroupBase);
+    uint32_t q_adicity =
+        ComputeAdicity(q, gmp::FromDecString(base::NumberToString(size_)));
```

Comment:
I think 2 string conversions are bad idea in terms of performance. Could you implement `ToMpzClass()` in `BigInt<N>` and copy limbs there and replace all these pattern with `ToMpzClass()`

Reply:
I've just refactored it to use `gmp::FromUnsignedInt` is it okay?